arch/arm/src/stm32/hardware/stm32_spi.h

```diff
@@ -135,7 +150,8 @@
 #define SPI_CR1_SSI               (1 << 8)  /* Bit 8: Internal slave select */
 #define SPI_CR1_SSM               (1 << 9)  /* Bit 9: Software slave management */
 #define SPI_CR1_RXONLY            (1 << 10) /* Bit 10: Receive only */
-#if defined(CONFIG_STM32_STM32F30XX) || defined(CONFIG_STM32_STM32F37XX)
+#if defined(CONFIG_STM32_STM32F30XX) || defined(CONFIG_STM32_STM32F37XX) \
```

Comment:
Should we not increase the ifdef rash? 

Using HAS_xxxxx macros and distribute them to the chip file?

The idea is a single line ifdef

`#if defined(STM32_HAS_SPI_CRC))`

This in the chip file define `#define STM32_HAS_SPI_CRC ` For all the chips that support it.

Reply:
Yes. I am trying to reduce the ifdef rash in this file. Already added HAVE_SPI_I2S and HAVE_SPI_TI_MODE. I'll add a STM32_HAS_SPI_CRC as well. Makes sense to put it in chip.h file.

Reply:
> The idea is a single line ifdef
> 
> `#if defined(STM32_HAS_SPI_CRC))`
> 
> Then in the chip file define `#define STM32_HAS_SPI_CRC ` For all the chips that support it.

On 2nd thought, are you sure the HAVE_* macros should go in chip.h? Looking at that file, it contains only number-of-peripherals macros like STM32_NGPIO, STM32_NDAC, etc. No HAVE_* macros for anything.

I have made the changes in stm32_spi.h and have not moved any HAVE_* macros to chip.h yet. Please see if this is acceptable. Thanks

Reply:
Look how it is done in Kinetis.  https://github.com/apache/incubator-nuttx/blob/master/arch/arm/include/kinetis/kinetis_mcg.h#L50 

That is the knot that pulls all the IP blocks together for the mgc. If you go up one level the other knots sim and pmc are there as well.  

We can do something like that for the STM32 IP blocks. 

Chip.h is the "differentiator" listing N of IP blocks. It would make sense to silo the version of IP the same way.  The SoC (chip) "has-a" set of IP that is as some revision. It will be used in other SoC. If you can name the feature then add the feature to the chip section in chip file. That is great. If you can roll it it up and label it.  STM32_SPI_Vn that that will work well to.

Reply:
On further study it seems that for STM32, the STM32_HAVE_IP_* are setup in Kconfig, for example, STM32_HAVE_IP_TIMERS_V1 and others. These are selected in nuttx/arch/arm/src/stm32/Kconfig based on the chip model.

Since this is the precedent for STM32, this PR should follow it.

If I roll it up and label it as v1, v2, v3, I'm trying to figure out which features are in which version...